### PR TITLE
Moves exclusive lists toggle to list options

### PR DESCRIPTION
**Pull request by hobgoblina**
Going off of this comment (https://github.com/glitch-soc/mastodon/pull/1996#issuecomment-1356170800), figured I'd see if I could get it working. Turns out doing it this way gives a bit of code reduction and can re-use the `SettingsToggle` css to match other feed toggles while also giving room for a slightly more descriptive label!

Screen Shot 2023-01-26 at 03 32 45
Screen Shot 2023-01-26 at 03 32 24

### CODE REVIEW

File: app/javascript/mastodon/features/list_timeline/index.js [hunk 1]
```diff
@@ -187,6 +195,13 @@ class ListTimeline extends React.PureComponent {
             </button>
           </div>
 
+          <div className='setting-toggle'>
+            <Toggle id='lists.exclusive' defaultChecked={isExclusive} onChange={this.onExclusiveToggle} />
+            <label htmlFor='lists.exclusive' className='setting-toggle__label'>
+              <FormattedMessage id='lists.exclusive' defaultMessage='Hide these toots from home' />
```
Comment (Plastikmensch) [inline app/javascript/mastodon/features/list_timeline/index.js]: I love the use of "toots" here, but upstream is desperate to remove "toot", so this wouldn't be accepted.
```suggestion
              <FormattedMessage id='lists.exclusive' defaultMessage='Hide these posts from home' />
```

**Review by Plastikmensch — CHANGES_REQUESTED** [review]
I haven't tested these changes yet, but overall it looks very similar to what I came up with before the glitch PR was closed, so it should work.

File: app/javascript/mastodon/features/list_timeline/index.js (hunk 1 above)
Comment (hobgoblina) [reply app/javascript/mastodon/features/list_timeline/index.js]: Heh, didn't know that. Toots waa finally starting to grow on me 😜

**Comment by dariusk** [comment]
This looks really good to me, in terms of how the feature is implemented! It's also going to be a really good reference for me going forward as to a less hacky way to implement this kind of thing.

**Pull request merged**